src/primitives.ts

```diff
@@ -927,4 +927,110 @@ export class Wire {
     }
     context.stroke();
   }
-}
\ No newline at end of file
+}
+
+
+export class PieChart {
+  pieParts: Array<PiePart>=[];
+  attributeNames: Attribute[]=[]; // kept for a future release but useless now
+
+  constructor(public tooltip:Tooltip,
+              public slicingVariable:string,
+              public dataSamples: any[],
+              public type_:string,
+              public name:string) {
+    this.definePieParts(dataSamples);
+  }
+
+  public static deserialize(serialized): PieChart {
+    let default_point_style = {color_fill: string_to_rgb('lightviolet'), color_stroke: string_to_rgb('lightgrey')};
+    let default_tooltip = {attributes: serialized['slicing_variable']};
+    let default_dict_ = {point_style:default_point_style, tooltip: default_tooltip};
+    serialized = set_default_values(serialized, default_dict_);
+    let tooltip = Tooltip.deserialize(serialized['tooltip']);
+    return new PieChart(tooltip,
+                        serialized['slicing_variable'],
+                        serialized['data_samples'],
+                        serialized['type_'],
+                        serialized['name']);
+  }
+
+  computeTotalSample(dataSamples: Object[]): number {
+    let sumSamples: number = 0;
+    let normedRatio: number = 2*Math.PI;
+
+    dataSamples.forEach(sample => {
+      sumSamples += sample[this.slicingVariable]
+    });
+    normedRatio /= sumSamples;
+    return normedRatio
+  }
+
+  definePieParts(dataSamples: Object[]): void {
+    let initAngle: number = Math.PI/2;
+    let nextAngle: number = initAngle;
+
+    let normedRatio = this.computeTotalSample(dataSamples);
+    dataSamples.forEach(sample => {
+      nextAngle -= sample[this.slicingVariable] * normedRatio;
+      let partI = new PiePart(0, 0, 10, initAngle, nextAngle, true);
+      this.pieParts.push(partI);
+      initAngle = nextAngle;
+    })
+  }
+}
+
+/**
+ * A class for drawing Pie parts.
+ */
+ export class PiePart{
+  hidden_color:string = '';
+  isMouseOver: boolean = false;
+  isSelected: boolean = false;
+  path: Path2D;
+  clicked: boolean = false;
+  color: string = '';
+
+  /**
+   * @param centerX the center x coordinate
+   * @param centerY the center y coordinate
+   * @param radius radius
+   * @param initAngle in radian
+   * @param endAngle in radian
+   * @param anticlockwise true or false whether you want the arc the be drawn anticlockwise or not.
+   */
+  constructor(public centerX:number,
```

Comment:
I think that we should ask ourself which variable has to be a PieChart variable, a PiePart variable, a value passed to the constructor but without being a class variable (remove the public in the constructor), a value passed to a specific method like draw() for example. 

Let discuss about it tomorrow